src/backend/api/meals.js

```diff
@@ -20,68 +21,83 @@ router.get("/", async (request, response) => {
-          const mealLimit = Number(request.query.limit);
-          const limitingNumber = await knex("meals").select("*").limit(mealLimit);
-          response.json(limitingNumber);
-        } else {
-          const titles = await knex("meals").select("title");
-          response.json(titles);
-        }
-      } catch (error) {
-        throw error;
-      });
-
-
-
-    router.post("/", async (req, res) => {
-      try {
-        const addMeal = await knex("meal").insert(req.body);
-        res.json(addMeal);
-      } catch (error) {
-        throw error;
-      }
-    });
-
-    router.get("/:id", async (req, res) => {
-      try {
-        const mealById = await knex("meal").where(req.params);
-        res.json(mealById);
-      } catch (error) {
-        throw error;
-      }
-    });
-
-    router.put("/:id", async (req, res) => {
-      try {
-        const updatedMealId = await knex("meal").where(req.params).update(req.body);
-        res.json(updatedMealId);
-      } catch (error) {
-        throw error;
-      }
-    });
-
-
-    router.delete("/:id", async (req, res) => {
-      try {
-        const deletedById = await knex("meal").where(req.params).del();
-        res.json(deletedById);
-      } catch (error) {
-        throw error;
-      }
-    });
-
-    module.exports = router;
\ No newline at end of file
+    if ("createdAfter" in request.query) {
+      const creationDate = new Date(request.query.createdAfter)
+      const newCreationDate = await knex("meals").where("created_date", ">=", creationDate)
+      response.json(newCreationDate)
+      return;
+    }
+
+    if ("limit" in request.query) {
+      const mealLimit = parseFloat(request.query.limit)
+      const limitingNumber = await knex("meals").limit(mealLimit)
+      response.json(limitingNumber)
+      return;
+    } else {
+      const meals = await knex("meals")
+      response.json(meals);
+    }
+
+  } catch (error) {
+    throw error;
+  }
+
+});
+
+
+router.post("/", async (request, response) => {
+  try {
+    const addMeal = await knex("meals").insert(request.body);
+    response.json(addMeal);
+  } catch (error) {
+    throw error;
+  }
+});
+
+router.get("/:id", async (request, response) => {
+  try {
+    const mealById = await knex("meals")
+      .where({
+        id: parseInt(request.params.id)
+      })
+    response.json(mealById)
+  } catch (error) {
+    throw error;
+  }
+});
+
+router.put("/:id", async (request, response) => {
+  try {
+    const updatedMealId = await knex("meals")
+      .where({
+        id: Number(request.params.id)
+      })
+      .update(request.body)
+    response.json(updatedMealId)
+  } catch (error) {
+    throw error;
+  }
+});
+
+router.delete("/:id", async (request, response) => {
+  try {
+    const deletedById = await knex("meals")
+      .where({
+        id: request.params.id
+      }).del()
+    response.json(deletedById)
+  } catch (error) {
+    throw error;
+  }
+});
+
+module.exports = router;
```

Comment:
It would be nice if we could have this code formatted so that it could be more readable.
Can you try to format it with Prettier or something like that?